crates/sui-core/src/checkpoints/checkpoint_executor/mod.rs

```diff
@@ -542,12 +545,23 @@ async fn execute_transactions(
     authority_state
         .database
         .epoch_store()
-        .insert_pending_certificates(&txns)?;
-
-    authority_state.transaction_manager.enqueue(txns).await?;
+        .insert_pending_certificates(&synced_txns)?;
 
     authority_state
-        .database
-        .notify_read_effects(tx_digests)
-        .await
+        .transaction_manager
+        .enqueue(synced_txns)
+        .await?;
+
+    // Once synced_txns have been awaited, all txns should have effects committed.
+    let effects_future = authority_state.database.notify_read_effects(all_tx_digests);
+    match timeout(LOCAL_EXECUTION_TIMEOUT, effects_future).await {
+        Err(_elapsed) => {
+            panic!(
```

Comment:
Overall this is great, but we don't want to panic on a timeout. There are lots of reasons this could take too long. Its okay to wait forever for this to finish (since we really have no alternative), although we should definitely log if it is taking longer than expected so that we can investigate.

Reply:
Will take care of in #6980